modules/apps/jax-rs/mdm-admin/src/main/java/org/wso2/carbon/mdm/api/User.java

```diff
@@ -784,4 +784,39 @@ public Response resetPassword(UserCredentialWrapper credentials) throws MDMAPIEx
             throw new MDMAPIException(errorMsg, e);
         }
     }
+
+    /**
+     * Method to change the user password.
+     *
+     * @param credentials Wrapper object representing user credentials.
+     * @return {Response} Status of the request wrapped inside Response object.
+     * @throws MDMAPIException
+     */
+    @POST
+    @Path("reset-password")
+    @Consumes({MediaType.APPLICATION_JSON})
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response resetPasswordByAdmin(UserCredentialWrapper credentials) throws MDMAPIException {
+        UserStoreManager userStoreManager = MDMAPIUtils.getUserStoreManager();
+        ResponsePayload responsePayload = new ResponsePayload();
+        try {
+            byte[] decodedNewPassword = Base64.decodeBase64(credentials.getNewPassword());
+            userStoreManager.updateCredentialByAdmin(credentials.getUsername(), new String(
+                    decodedNewPassword, "UTF-8"));
+            responsePayload.setStatusCode(HttpStatus.SC_CREATED);
+            responsePayload.setMessageFromServer("User password by username: " + credentials.getUsername() +
+                    " was successfully changed.");
+            return Response.status(HttpStatus.SC_CREATED).entity(responsePayload).build();
+        } catch (UserStoreException e) {
+            String errorMsg = "Exception in trying to change the password by username: " + credentials.getUsername();
+            log.error(errorMsg, e);
+            responsePayload.setStatusCode(HttpStatus.SC_BAD_REQUEST);
```

Comment:
Maybe a better option would be to go with the builder pattern where you can construct this response with a single liner!.